mesa/model.py

```diff
@@ -21,7 +22,16 @@ class Model:
     def __new__(cls, *args: Any, **kwargs: Any) -> Any:
         """Create a new model object and instantiate its RNG automatically."""
         obj = object.__new__(cls)
-        obj._seed = kwargs.get("seed", None)
+        obj._seed = kwargs.get("seed")
+        if obj._seed is None:
+            # See https://docs.python.org/3/library/random.html#random.seed
+            # which says "If a is omitted or None, the current system time is used."
+            # We explicitly specify the seed here so that we know its value.
+            # datetime.now().timestamp() is more cross-platform than using
+            # time.time(), and guarantees that the precision is to the
+            # microsecond.
+            current_timestamp = datetime.datetime.now(datetime.timezone.utc).timestamp()
```

Comment:
I don't understand why we need that additional precision here - time.time seems to be simpler and should in any case be sufficient for just having a known seed.

Reply:
See https://stackoverflow.com/questions/27276135/python-random-system-time-seed#comment134270695_27276285.

Reply:
Well at the beginning of https://docs.python.org/3/library/time.html#module-time it says that time() will always return the most accurate time available. There is simply no logical way for datetime to achieve a higher precision. This might have been different in the past, the answer is from 2014

Reply:
The answer is from 2014, but that comment is from 2023.

From https://docs.python.org/3/library/time.html#time.time

> Note that even though the time is always returned as a floating point number, not all systems provide time with a better precision than 1 second. While this function normally returns non-decreasing values, it can return a lower value than a previous call if the system clock has been set back between the two calls.

Reply:
Yes, but if the operating system doesn't provide a better accuracy, how should datetime have a higher precision?

Reply:
The datetime, by specification, with the timestamp, will outputs the microseconds.

Reply:
Yes, but having the milliseconds (which are all 0 if no higher precision is available) provides no advantage. Why should a number with trailing zeros be better suitable as a random seed. What's your train of thought here?

Reply:
To make sure that simulations that are launched in parallel (that are within a second from each other) will have different seeds.

Reply:
The milliseconds/microseconds being 0's should only happen with `time.time()`.

Reply:
But the precision of datetime can't be better than that of time.time()! That's what I keep trying to tell you. 

There is just a difference between the display of your time. If you run on a machine with only second precision time.time() might for example return the value `1234`. 

If you run datetime.now() the datetime module will receive the same value - `1234`. The OS simply doesn't know better. But the  datetime is *required* to give you milliseconds. So it will return you `[...]34.000`. But you still end up with the same value within each second.

Reply:
The elaborate SO answer wouldn't be worth it if it were to return 123.000000, but anyway, `random.random()` is the simplest way to go for it, hence, case closed.

Reply:
You demonstrated enough times that you are very capable of logical thinking. I don't know why you keep refusing to use that ability. I said or asked several times now: if the OS doesn't provide higher precision where should the milliseconds come from? There is no logical way to derive them. I'll add a comment on SO

Reply:
Why did you assume that the OS can't provide higher precision with `datetime.now()`? It's just that `time.time()` and `datetime.now()` may output different precision, depending on the system. Don't just conclude on your own whim that your assumption of the inner workings of `time.time()` and `datetime.now()` were correct in the first place.

Reply:
Well I worked a lot with datetimes and once with an embedded system that had no subsecond precision. Not at the same time, but I think it's fine to say my understanding is a bit more than a "whim". Plus,as I quoted, it says in front of the time docs that time() uses the best precision available. 

But I definitely agree that we can quit this conversation now. I just wanted to help you understand.

Reply:
https://docs.python.org/3/library/datetime.html#datetime.datetime.now

> If optional argument tz is None or not specified, this is like [today()](https://docs.python.org/3/library/datetime.html#datetime.datetime.today), but, if possible, supplies more precision than can be gotten from going through a [time.time()](https://docs.python.org/3/library/time.html#time.time) timestamp (for example, this may be possible on platforms supplying the C gettimeofday() function).

Is this not clear enough?

Reply:
I'm fine with moving on. I didn't see your last message before my comment.

Reply:
Please let's talk considerately to each other. I am going to resolve this conversation